File: frontend/app/src/main/java/com/example/gameon/classes/Game.kt
package com.example.gameon.classes

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue

@Parcelize
data class Game(
    val game_id: Int? = null, // generated on server-side
    val game_name: String,
    val description: String,
    val group_size: Int,
    val groups: List<Group>? = null
): Parcelable

File: backend/src/entity/Game.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from "typeorm";
import { Group } from "./Group";

@Entity()
export class Game {
  @PrimaryGeneratedColumn()
  game_id!: number;

  @Column({ unique: true })
  game_name!: string;

  @Column({ nullable: true })
  description!: string;

  @Column({ default: 2})
  group_size!: number;

  @OneToMany(() => Group, (group) => group.game)
  groups!: Group[];
}